modules/develop/pages/produce-data/leader-pinning.adoc

```diff
@@ -0,0 +1,39 @@
+// tag::single-source[]
+:description: Learn about leader pinning and how to configure a preferred partition leader based on availability zones.
+
+Produce requests to Redpanda topics go through the topic partition leader, which then ensure that messages are synced across its follower replicas. For a Redpanda cluster deployed across multiple availability zones (AZs), leader pinning is a mechanism to ensure that a topic's partition leaders are geographically closer to clients. Similar to xref:develop:consume-data/follower-fetching.adoc[follower fetching] for consume requests, leader pinning can lower networking costs and help guarantee lower latency by routing produce requests to brokers located in certain AZs.
+
+== Prerequisites
+
+include::shared:partial$enterprise-license.adoc[]
+
+You must xref:manage:rack-awareness.adoc#configure-rack-awareness[configure rack awareness] for the cluster.
+
+== Configure leader pinning
+
+The following steps assume that you have Redpanda deployed in a multi-AZ setup, and you have configured each broker so that the config_ref:rack,true,properties/broker-properties[] configuration property contains rack IDs corresponding to the AZ IDs.
+
+. Set the topic configuration property `redpanda.leaders.preference`. The property accepts the following string values:
++
+** `none` (default): Opt out the topic explicily from leader pinning.
+** `racks:<rack1>[,<rack2>,...]`: Sets the preferred location (rack) of partition leaders to brokers with rack IDs in this list. The list can contain one or more rack IDs. 
++
+You can list the IDs in any order. Spaces in the list are ignored, i.e. `racks:rack1,rack2` and `racks: rack1,  rack2` are equivalent, but you cannot specify empty rack IDs, for example, `racks: rack1,,rack2`.
+
+. Set the cluster configuration property `default_leaders_preference`. This property sets the default leader pinning configuration for all topics that don’t have `redpanda.leaders.preference` explicitly set. It accepts values in the same format as `redpanda.leaders.preference`. Default: `none` 
++
+This property also affects internal topics such as __consumer_offsets and transaction coordinators.
+
+NOTE: If the config_ref:enable_rack_awareness,true,properties/cluster-properties[] cluster configuration property is set to `false`, leadership pinning is disabled across the cluster.
+
+If there is more than one broker in the preferred AZ (or AZs), leader pinning distributes partition leaders uniformly across brokers in the AZ.
+
+== Leader pinning failover across availability zones
+
+Suppose there are 3 AZs: A, B, and C, and A becomes unavailable. The failover behavior is as follows:
+
+* A topic with "A" as the preferred leader AZ will have its partition leaders uniformly distributed across B and C.
+* A topic with "A,B" as the preferred leader AZs will have its partition leaders in B.
+* A topic with “B” as the preferred leader AZ will have its partition leaders in B as well.
```

Comment:
```suggestion
* A topic with “B” as the preferred leader AZ will have its partition leaders in B as well.
```
```suggestion
* A topic with “B” as the preferred leader AZ has its partition leaders in B as well.
```
I line with style guide.

Reply:
@asimms41 Yeah, I wasn't too sure about this one. 50/50 on it, but I went with the future tense to try and emphasize the change that happens in the event of a failover (to me, "has its partition leaders" can sound like a description of the current state). WDYT?

Reply:
If that is more accurate then I'm happy to go with it.